loaders/httpfs/loader.go

```diff
@@ -0,0 +1,30 @@
+package httpfs
+
+import (
+	"io"
+	"net/http"
+)
+
+type httpFileSystemLoader struct {
+	http.FileSystem
+}
+
+func NewLoader(fs http.FileSystem) httpFileSystemLoader {
```

Comment:
Nice, https://golang.org/doc/effective_go.html#generality explains it quite well. So it's really about "unexported" structs only, I get it now. Thank you for sharing! 👍

I'll push another commit in a bit.